packages/plugin-ext/src/main/browser/terminal-main.ts

```diff
@@ -95,20 +95,22 @@ export class TerminalServiceMainImpl implements TerminalServiceMain, TerminalLin
     protected async trackTerminal(terminal: TerminalWidget): Promise<void> {
         let name = terminal.title.label;
         this.extProxy.$terminalCreated(terminal.id, name);
+
         const updateTitle = () => {
             if (name !== terminal.title.label) {
                 name = terminal.title.label;
                 this.extProxy.$terminalNameChanged(terminal.id, name);
             }
         };
-        terminal.title.changed.connect(updateTitle);
-        this.toDispose.push(Disposable.create(() => terminal.title.changed.disconnect(updateTitle)));
 
         const updateProcessId = () => terminal.processId.then(
             processId => this.extProxy.$terminalOpened(terminal.id, processId, terminal.terminalId, terminal.dimensions.cols, terminal.dimensions.rows),
             () => {/* no-op */ }
         );
+
         updateProcessId();
+        terminal.title.changed.connect(updateTitle);
```

Comment:
This looks like some code was moved around with no functional changes. Am I missing something here?

Reply:
The ordering here was important. Its been a while since I was in this code, but I think `updateProcessId()` needed to be called first to open the terminal and guarantee the IDs. With the changed event earlier in the code a race condition existed.

Reply:
But we're not even waiting for anything here: so while this might change the timing, it doesn't seem to guarantee any ordering of events, as far as I can tell. Since the `terminalOpened` waits for `terminal.processId`, there is no guarantee that `updateTitle` is not called before before `updateProcessId`. If the ordering dependencies are that subtle, we should at least add a comment, lest the next person reorders the code.

Reply:
I re-tested this and it seems this wasn't needed. Perhaps I just preferred the code that way!. Have removed the changes in this file.